### ISSUE TITLE
[Bug report] build failed with vite, fallback to webpack failed with sass-loader

### ISSUE THREAD

**Issue by hschaefer123** [vuepress/vuepress-next#970]
### Description

Hi,
just tried to migrate from beta 38 to 48.

Now, the preview server works even with my web components, but when i try to build docs, i got an error, that the relevant web components could not be found.

Like previously i tried to fallback to my former working webpack, but starting the dev server, i ran into issue, that the sass-loader could not be found.

```
info Initializing VuePress and preparing data...
✔ Compilation finished in 7314ms
success VuePress webpack dev server is listening at http://localhost:9090/
ERROR in ./node_modules/@vuepress/theme-default/lib/client/config.js 5:0-29
Module not found: Error: Can't resolve 'sass-loader' in 'C:\VSC\@udina\udina.docng48'

ERROR in ./docs/.vuepress/components/HeroBackground.vue 5:0-86
Module not found: Error: Can't resolve 'sass-loader' in 'C:\VSC\@udina\udina.docng48'

webpack compiled with 2 errors
```

Any idea to get this fixed?

webpack is ok for me (if i got it running again with the new release).

Regards
Holger

### Reproduction

no repo avail

### Used Package Manager

npm

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (12) x64 AMD Ryzen 5 3600 6-Core Processor
    Memory: 5.85 GB / 15.95 GB
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Utilities:
    Git: 2.35.1.
  Browsers:
    Chrome: 103.0.5060.66
    Edge: Spartan (44.22000.120.0), Chromium (103.0.1264.37)
  npmPackages:
    @vuepress/bundler-vite:  2.0.0-beta.48 
    @vuepress/bundler-webpack: Not Found
    @vuepress/cli:  2.0.0-beta.48 
    @vuepress/client:  2.0.0-beta.48 
    @vuepress/core:  2.0.0-beta.48 
    @vuepress/markdown:  2.0.0-beta.48 
    @vuepress/plugin-active-header-links:  2.0.0-beta.48 
    @vuepress/plugin-back-to-top:  2.0.0-beta.48
    @vuepress/plugin-container:  2.0.0-beta.48
    @vuepress/plugin-docsearch: Not Found
    @vuepress/plugin-external-link-icon:  2.0.0-beta.48
    @vuepress/plugin-git: ^2.0.0-beta.48 => 2.0.0-beta.48
    @vuepress/plugin-google-analytics: Not Found
    @vuepress/plugin-medium-zoom:  2.0.0-beta.48
    @vuepress/plugin-nprogress:  2.0.0-beta.48
    @vuepress/plugin-palette:  2.0.0-beta.48
    @vuepress/plugin-prismjs:  2.0.0-beta.48
    @vuepress/plugin-pwa: ^2.0.0-beta.48 => 2.0.0-beta.48
    @vuepress/plugin-pwa-popup: ^2.0.0-beta.48 => 2.0.0-beta.48
    @vuepress/plugin-register-components: ^2.0.0-beta.48 => 2.0.0-beta.48
    @vuepress/plugin-search: ^2.0.0-beta.48 => 2.0.0-beta.48
    @vuepress/plugin-shiki: Not Found
    @vuepress/plugin-theme-data:  2.0.0-beta.48
    @vuepress/plugin-toc: Not Found
    @vuepress/shared:  2.0.0-beta.48
    @vuepress/theme-default:  2.0.0-beta.48
    @vuepress/utils:  2.0.0-beta.48
    vue:  3.2.37
    vue-loader: Not Found
    vue-router:  4.0.16
    vuepress: ^2.0.0-beta.48 => 2.0.0-beta.48
    vuepress-vite:  2.0.0-beta.48
    vuepress-webpack: Not Found
```

**Comment by hschaefer123** [comment]
Hi,
i was able to fix the issue using

```
npm install sass-loader -D
```

Maybe the deps can  be added to vuepress-webpack package.

Also compiling works again, so i will deloy it and see, if everything is working as prevciously.

Regards
Holger

**Comment by meteorlxy** [comment]
> Maybe the deps can be added to vuepress-webpack package.

That's a good idea. I used to thought that sass-loader is only a dep of default theme, which might not be useful in other themes. But yes, as default theme is a dep of vuepress-webpack, then sass-loader should also be included.

In fact, we have listed sass-loader in the peer-deps of default theme. But obviously we should also mention it in the docs.

https://github.com/vuepress/vuepress-next/blob/837738136523cec1672ad9619f806ec6fb180b13/packages/%40vuepress/theme-default/package.json#L52-L59
